From 82a8a10b14755ec06bf95e86675c327ced4177c5 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Wed, 21 Aug 2024 18:20:55 +0200
Subject: [PATCH 1074/1695] drivers: stm32_cpu_opp: export
 opp_voltage_is_supported function

Export opp_voltage_is_supported() function to use it in future driver.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 34693664b3767e93e8dc73d1d125a604ec7c21ee drivers: stm32_cpu_opp: export opp_voltage_is_supported function]
Change-Id: Ibbfddfcbe4cd42cbbc0db79970928a012cbd9a67
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/398309
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Patrice GUILEMOND <patrice.guilemond-ext@st.com>
---
 core/drivers/stm32_cpu_opp.c         | 2 +-
 core/include/drivers/stm32_cpu_opp.h | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/drivers/stm32_cpu_opp.c b/core/drivers/stm32_cpu_opp.c
index 7c385d152..e6c23f251 100644
--- a/core/drivers/stm32_cpu_opp.c
+++ b/core/drivers/stm32_cpu_opp.c
@@ -108,7 +108,7 @@ static TEE_Result opp_set_voltage(struct regulator *regul, int volt_uv)
  * the function may adjust the input parameter volt_uv to a higher
  * supported value and still return true.
  */
-static bool opp_voltage_is_supported(struct regulator *regul, uint32_t *volt_uv)
+bool opp_voltage_is_supported(struct regulator *regul, uint32_t *volt_uv)
 {
 	const int target_volt_uv = *volt_uv;
 	int new_volt_uv = 0;
diff --git a/core/include/drivers/stm32_cpu_opp.h b/core/include/drivers/stm32_cpu_opp.h
index f50a30a92..5a19e6d13 100644
--- a/core/include/drivers/stm32_cpu_opp.h
+++ b/core/include/drivers/stm32_cpu_opp.h
@@ -16,6 +16,8 @@ struct clk;
 size_t stm32_cpu_opp_count(void);
 #endif
 
+bool opp_voltage_is_supported(struct regulator *regul, uint32_t *volt_uv);
+
 /* Get level value identifying CPU operating point @opp_index */
 unsigned int stm32_cpu_opp_level(size_t opp_index);
 
-- 
2.47.2

